src/twisted/internet/test/test_fdset.py

```diff
@@ -115,6 +115,19 @@ def fail():
         # Getting here means the right thing happened probably.
 
 
+    def test_removeReader_selectables_nokey(self):
+        """
+        L{reactor.removeReader()} accepts an L{IReadDescriptor} provider
+        previously passed to C{reactor.addReader()}. Checks if it doesn't
+        raise KeyError exception
+        """
+        reactor, fd, server = self._simpleSetup()
+        reactor.addReader(fd)
+        reactor._selectables = {}
```

Comment:
Can you reproduce the problem?  I don't think most people encounter this when using epollreactor.  That suggests the bug isn't that epollreactor just randomly and arbitrarily forgets about file descriptors sometimes - but that something about your application/environment/usage provokes this bug.  If we could understand what it is that does that, it might give us some clues about what's wrong - and thus suggest a better test.